actions/views/rhs.ts

```diff
@@ -251,6 +281,46 @@ export function showPinnedPosts(channelId?: string) {
     };
 }
 
+export function showChannelFiles(channelId?: string) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState();
+        const currentChannelId = getCurrentChannelId(state);
+        const teamId = getCurrentTeamId(state);
+
+        dispatch(batchActions([
+            {
+                type: ActionTypes.UPDATE_RHS_STATE,
+                channelId: channelId || currentChannelId,
+                state: RHSStates.CHANNEL_FILES,
+            },
+        ]));
+
+        const results: any = await dispatch(performSearch('channel:' + (channelId || currentChannelId)));
+
+        let data: any;
+        if (results && results.length === 2 && 'data' in results[1]) {
+            data = results[1].data;
+        }
+
+        dispatch(batchActions([
+            {
+                type: SearchTypes.RECEIVED_SEARCH_POSTS,
+                data,
+            },
+            {
+                type: SearchTypes.RECEIVED_SEARCH_TERM,
```

Comment:
The use of `RECEIVED_*` between these two cases seems unexpected -- I'm imagining the former contains the results of a query to the server, whereas the latter seems like purely local state. Should the action type be renamed to reflect this?

Reply:
Nope as this PR, this is already complex enough and we are talking about a lot of places where it should be changed, for example for others like `showPinnedPosts`